skglm/experimental/reweighted.py

```diff
@@ -0,0 +1,44 @@
+import numpy as np
+from numpy.linalg import norm
+from skglm.estimators import Lasso
+
+
+class ReweightedLasso(Lasso):
+    def __init__(self, alpha=1., n_reweight=5, reweight_penalty=None, *args, **kwargs):
+        super().__init__(alpha=alpha, *args, **kwargs)
+        self.n_reweight = n_reweight
+        self.reweight_penalty = reweight_penalty or self._reweight_penalty
```

Comment:
in sklearn estimator's __init__ we're only allowed to do self.foo = foo, nothing more. All other stuff must happen at fit time